WBooks/AddNew/AddNewController.swift

```diff
@@ -7,23 +7,165 @@
 //
 
 import UIKit
+import ReactiveSwift
+import ReactiveCocoa
+import Result
+import WolmoCore
 
 class AddNewController: UIViewController {
-
+    
+    private let formFilled = MutableProperty(false)
+    
+    private var loading = false
+    
+    private lazy var _view: AddNewView = AddNewView.loadFromNib()!
+    
+    private var _viewModel: AddNewViewModel = AddNewViewModel()
+    
+    override func loadView() {
+        view = _view
+    }
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-
-        // Do any additional setup after loading the view.
+        _view.setView()
+        setBindings()
+        setFormValidation()
+        setSubmitButton()
+        setImagePicker()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        configureNavBar()
+    }
+    
+    private func setBindings() {
+        _viewModel.bookName <~ _view.booksName.reactive.continuousTextValues.skipNil()
+        _viewModel.author <~ _view.author.reactive.continuousTextValues.skipNil()
+        _viewModel.year <~ _view.year.reactive.continuousTextValues.skipNil()
+        _viewModel.genre <~ _view.genre.reactive.continuousTextValues.skipNil()
+        _viewModel.descrpition <~ _view.bookDescription.reactive.continuousTextValues.skipNil()
+    }
+    
+    private func setFormValidation() {
+        let bookNameValidation = _view.booksName.reactive.continuousTextValues.skipNil().map { text in
+            !text.isEmpty
+        }
+        let authorValidation = _view.author.reactive.continuousTextValues.skipNil().map { text in
+            !text.isEmpty
+        }
+        let yearValidation = _view.year.reactive.continuousTextValues.skipNil().map {text in
+            !text.isEmpty
+        }
+        let genreValidation = _view.genre.reactive.continuousTextValues.skipNil().map {text in
+            !text.isEmpty
+        }
+        let descriptionValidation = _view.bookDescription.reactive.continuousTextValues.skipNil().map {text in
+            !text.isEmpty
+        }
+        let isButtonEnabled = Signal<Bool, NoError>.combineLatest(bookNameValidation, authorValidation, yearValidation, genreValidation, descriptionValidation).map {
+            $0 && $1 && $2 && $3 && $4
+        }
+        
+        isButtonEnabled.observeValues { [weak self] value in
+            self?.formFilled.value = value
+        }
+    }
+    
+    private func setSubmitButton() {
+        formFilled.producer.startWithValues { [weak self] formFilled in
+            if let this = self {
+                if formFilled && !this.loading {
+                    this._view.enableSubmit()
+                } else {
+                    this._view.disableSubmit()
+                }
+            }
+        }
+        _view.submitButton.reactive.controlEvents(.touchUpInside).observeValues { [weak self] _ in
+            self?.loading = true
```

Comment:
You could handle this with reactive swift in a simpler way. If you want, you can search it, but it's fine if you leave this in training.